main/acle.md

```diff
@@ -9546,25 +9680,2141 @@ possible to write these operations using normal C arithmetic. For example:
+  svfloat32x2_t svcvt_f32[_s32_x2](svint32x2_t zn);
+  
+  
+  // Variants are also available for _f32[_u32_x4], _s32[_f32_x4] and _u32[_f32_x4]
+  __attribute__((arm_streaming))
+  svfloat32x4_t svcvt_f32[_s32_x4](svint32x4_t zn);
+  
+  ```
+
+#### SQCVT, SQCVTU, UQCVT
+
+Multi-vector saturating extract narrow
+
+``` c
+  // Variants are also available for _u16[_s32_x2] and _u16[_u32_x2]
+  __attribute__((arm_streaming))
+  svint16_t svqcvt_s16[_s32_x2](svint32x2_t zn);
+  
+  
+  // Variants are also available for _u8[_s32_x4], _u8[_u32_x4], _s16[_s64_x4],
+  // _u16[_s64_x4] and _u16[_u64_x4]
+  __attribute__((arm_streaming))
+  svint8_t svqcvt_s8[_s32_x4](svint32x4_t zn);
+  
+  ```
+
+#### SQCVTN, SQCVTUN, UQCVTN
+
+Multi-vector saturating extract narrow and interleave
+
+``` c
+  // Variants are also available for _u16[_s32_x2] and _u16[_u32_x2]
+  __attribute__((arm_streaming_compatible))
+  svint16_t svqcvtn_s16[_s32_x2](svint32x2_t zn);
+  
+  
+  // Variants are also available for _u8[_s32_x4], _u8[_u32_x4], _s16[_s64_x4],
+  // _u16[_s64_x4] and _u16[_u64_x4]
+  __attribute__((arm_streaming))
+  svint8_t svqcvtn_s8[_s32_x4](svint32x4_t zn);
+  
+  ```
+
+#### UDOT, SDOT, FDOT (vectors)
+
+Multi-vector dot-product (2-way and 4-way)
+
+``` c
+  // Variants are also available for _s32_s16_s16 and _u32_u16_u16
+  __attribute__((arm_streaming_compatible))
+  svfloat32_t svdot[_f32_f16_f16](svfloat32_t zda, svfloat16_t zn, svfloat16_t zm);
+  
+  ```
+
+#### UDOT, SDOT, FDOT (indexed)
+
+Multi-vector dot-product (2-way and 4-way)
+
+``` c
+  // Variants are also available for _s32 and _u32
+  __attribute__((arm_streaming_compatible))
+  svfloat32_t svdot_lane[_f32](svfloat32_t zda, svfloat16_t zn, svfloat16_t zm,
+                               uint64_t imm_idx);
+  
+  ```
+
+#### FDOT, BFDOT, SUDOT, USDOT, SDOT, UDOT (store into ZA, single)
+
+Multi-vector dot-product (2-way and 4-way)
+
+``` c
+  // Variants are available for:
+  //   _za32[_bf16]
+  //   _za32[_f16]
+  //   _za32[_s8]
+  //   _za32[_s16]
+  //   _za32[_u8]
+  //   _za32[_u16]
+  //   _za64[_s16] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  //   _za64[_u16] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svdot[_single]_za32[_bf16]_vg1x2(uint32_t slice, svbfloat16x2_t zn,
+                                        svbfloat16_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_bf16]
+  //   _za32[_f16]
+  //   _za32[_s8]
+  //   _za32[_s16]
+  //   _za32[_u8]
+  //   _za32[_u16]
+  //   _za64[_s16] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  //   _za64[_u16] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svdot[_single]_za32[_bf16]_vg1x4(uint32_t slice,
+                                        svbfloat16x4_t zn, svbfloat16_t zm);
+  
+  
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsudot[_single]_za32[_s8]_vg1x2(uint32_t slice, svint8x2_t zn,
```

Comment:
I suppose this raises a policy decision: do we want to continue to encode sign differences in the mnemonic, as for Advanced SIMD and SVE?  Or do we want to use suffixes?  If we want to use the mnemonic, there's probably no need for three type suffixes on line 9925.  Just two should be enough (which would make the `_n` forms feasible again).  If instead we switch to using suffixes, then we should probably do the same here too (`svdot[_single]_za32[_s8_u8]_vg1x2`).